src/environment.js

```diff
@@ -56,21 +56,24 @@ function deriveGetBinding(realm: Realm, binding: Binding) {
   return realm.generator.deriveAbstract(types, values, [], createOperationDescriptor("GET_BINDING", { binding }));
 }
 
-export function havocBinding(binding: Binding): void {
-  let realm = binding.environment.realm;
+export function materializeBinding(realm: Realm, binding: Binding): void {
+  let realmGenerator = realm.generator;
   let value = binding.value;
+  if (value !== undefined && realmGenerator !== undefined && value !== realm.intrinsics.undefined)
```

Comment:
An initialized binding should never have an undefined value. I don't see how we can validly call materializeBinding before initialization is complete. Also a binding that has been initialized to realm.intrinsics.undefined seems to me like it should be materialized.